### PR TITLE
Fix for #322

### PR DESCRIPTION
**Pull request by matthewdavidson**
This PR is by no means complete - I've spent a bit of time familiarising myself with the codebase and this is my quick hack to get things working. However as a result I have some observations...

Whilst I guess it's relatively uncommon, lodash is a good example of a package that exposes entry points multiple levels deep such as `lodash/fp/object` or `lodash/fp/curryN`. The method currently used to resolve packages - by reading the components `package.json` - wont be enough to future proof this use case as it's not feasible to statically analyse the package for all of the possible entry points (at n levels of nesting).

An alternative approach would be to refactor the `require-wrapper` to pass any paths through to `require` as long as they match dependencies that are listed in the components `package.json` as well as conforming to the registries whitelist of dependencies.

@matteofigus thoughts?

### CODE REVIEW

**Comment by matteofigus** [comment]
Hi @matthewdavidson thanks for the PR. I agree about your proposal.

Summarising, I would be happy if both component and registry maintainer declare "lodash" as dependency, and then server.js is able to require anything that matches "lodash" or "lodash/anything" or "lodash/anything/anything(etc)...".

**Comment by matthewdavidson** [comment]
Apologies for the delay - I've updated the PR. 

`dependency-resolver.js` has been ditched in favor of `require-wrapper.js` serving only to validate any requires against the dependency whitelist.

File: npm-shrinkwrap.json [hunk 1]
```diff
@@ -3066,6 +2989,11 @@
         }
       }
     },
+    "require-package-name": {
+      "version": "2.0.1",
+      "from": "require-package-name@latest",
+      "resolved": "https://registry.npmjs.org/require-package-name/-/require-package-name-2.0.1.tgz"
+    },
```
Comment (matthewdavidson) [inline npm-shrinkwrap.json]: Done (i think).

Comment (matteofigus) [inline npm-shrinkwrap.json]: Thanks, can you just pull in the latest master so we can get that extra test ;)

**Comment by matteofigus** [comment]
Thanks for taking care of this @matthewdavidson - just published this to a new version

**Comment by matthewdavidson** [comment]
👍 🎉

**Pull request merged**